app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = current_user.comments.new(post_id: @post.id, author_id: current_user.id, text: comment_params)
+    @comment.post_id = @post.id
+
+    if @comment.save
+      redirect_to user_post_path(current_user, @post.id)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
```

Comment:
- [ ] I am unable to create a comment. From the image, you can see that your params do not contain the `post_id` value. Kindly check your routes. I think you need to wrap the comments routes within the posts one to get the `post_id`, instead of having it as a stand-alone routes, then refactor your view if necessary

  ```
  resources :users, only: [:index, :show] do
    resources :posts do
      resources :comments, only: [:new, :create]
  ...
  ```

  ![ffm](https://user-images.githubusercontent.com/36542986/193430193-5ea7545a-a0e0-4ba1-8830-01659d71dac0.jpg)